src/main/groovy/com/palantir/gradle/docker/PalantirDockerPlugin.groovy

```diff
@@ -94,6 +98,15 @@ class PalantirDockerPlugin implements Plugin<Project> {
                     buildCommandLine.addAll('--build-arg', "${buildArg.getKey()}=${buildArg.getValue()}")
                 }
             }
+            if (!ext.labels.isEmpty()) {
+                for (Map.Entry<String, String> label : ext.labels.entrySet()) {
+                    if (!label.getKey().matches(labelKeyPattern)) {
+                        throw new GradleException(String.format("Docker label '%s' contains illegal characters. " +
+                                "Label keys must only contain lowercase alphanumberic, `.`, or `-` characters.", label.getKey()))
```

Comment:
prefer defining the pattern as a string so can log the explicit expected pattern here.